src/GeniusYield/TxBuilder/Class.hs

```diff
@@ -190,22 +211,134 @@ class MonadError GYTxMonadException m => GYTxQueryMonad m where
     -- | Log a message with specified namespace and severity.
     logMsg :: HasCallStack => GYLogNamespace -> GYLogSeverity -> String -> m ()
 
--- | Class of monads for querying monads as a user.
-class GYTxQueryMonad m => GYTxMonad m where
+-- | Class of monads for querying special chain data.
+{- Note [Necessity of 'GYTxSpecialQueryMonad' and transaction building as a class method]
 
+The only purpose of 'GYTxSpecialQueryMonad' is to provide necessary information for building
+transactions. Since the inclusion of 'submitTx' under 'GYTxMonad', it is necessary to be able to
+build transactions within 'GYTxMonad'. Our current tx building interface requires these pieces
+of information. So this is a superclass to 'GYTxMonad'.
+
+However, transaction building _could_ be included as a class method of 'GYTxMonad'. But it is difficult
+to decide where to draw the line regarding the interface. Our transaction building mechanisms are aware of
+coin selection strategy, parallel transactions, chaining transactions etc. Should all this really be included
+under the class method in question?
+-}
```

Comment:
Important note around a potentially unnecessary monad and a prompt on reworking transaction building API